src/reporting/Market.sol

```diff
@@ -0,0 +1,653 @@
+    uint256 private constant AUTOMATED_REPORTER_DISPUTE_BOND_AMOUNT = 11 * 10**20;
+    uint256 private constant LIMITED_REPORTERS_DISPUTE_BOND_AMOUNT = 11 * 10**21;
+    uint256 private constant ALL_REPORTERS_DISPUTE_BOND_AMOUNT = 11 * 10**22;
+
+    uint256 private constant MIN_NUM_OUTCOMES = 2;
+    uint256 private constant MIN_PAYOUT_DENOMINATOR = 2;
+    uint256 private constant MAX_FEE_PER_ETH_IN_ATTOETH = 5 * 10 ** 17;
+    int256 private constant DISPLAY_PRICE_MIN = -(2 ** 254);
+    int256 private constant DISPLAY_PRICE_MAX = 2 ** 254;
+    uint256 private constant SET_COST_MULTIPLIER_MAX = 2 ** 254;
+    uint256 private constant APPROVAL_AMOUNT = 2 ** 254;
+    uint256 private constant AUTOMATED_REPORTING_DURATION_SECONDS = 3 days;
+    uint256 private constant AUTOMATED_REPORTING_DISPUTE_DURATION_SECONDS = 3 days;
+    address private constant NULL_ADDRESS = address(0);
+
+    ReportingWindow private reportingWindow;
+    uint256 private endTime;
+    uint8 private numOutcomes;
+    uint256 private marketCreationBlock;
+    bytes32 private topic;
+    address private automatedReporterAddress;
+    mapping(bytes32 => ReportingToken) private reportingTokens;
+    Cash private denominationToken;
+    address private creator;
+    IShareToken[] private shareTokens;
+    uint256 private finalizationTime;
+    bool private automatedReportReceived;
+    bytes32 private tentativeWinningPayoutDistributionHash;
+    bytes32 private finalPayoutDistributionHash;
+    DisputeBondToken private automatedReporterDisputeBondToken;
+    DisputeBondToken private limitedReportersDisputeBondToken;
+    DisputeBondToken private allReportersDisputeBondToken;
+    uint256 private validityBondAttoeth;
+    uint256 private automatedReporterBondAttoeth;
+
+    function initialize(ReportingWindow _reportingWindow, uint256 _endTime, uint8 _numOutcomes, uint256 _payoutDenominator, uint256 _feePerEthInAttoeth, Cash _denominationToken, address _creator, int256 _minDisplayPrice, int256 _maxDisplayPrice, address _automatedReporterAddress, bytes32 _topic) public payable beforeInitialized returns (bool _success) {
+        endInitialization();
+        require(address(_reportingWindow) != NULL_ADDRESS);
+        require(_payoutDenominator >= MIN_PAYOUT_DENOMINATOR);
+        require(_numOutcomes >= MIN_NUM_OUTCOMES);
+        // payoutDenominator must be a multiple of numOutcomes so we can evenly split complete set share payout on indeterminate
+        require((_payoutDenominator % _numOutcomes) == 0);
+        require(feePerEthInAttoeth <= MAX_FEE_PER_ETH_IN_ATTOETH);
+        require((DISPLAY_PRICE_MIN <= _maxDisplayPrice) && (_maxDisplayPrice <= DISPLAY_PRICE_MAX));
+        require((DISPLAY_PRICE_MIN <= _minDisplayPrice) && (_minDisplayPrice <= DISPLAY_PRICE_MAX));
+        require((1 <= uint256(_maxDisplayPrice.sub(_minDisplayPrice))) && (uint256(_maxDisplayPrice.sub(_minDisplayPrice)) <= SET_COST_MULTIPLIER_MAX));
```

Comment:
This check can probably be split into multiple checks for simplicity.  If we have space on the stack, I recommend an intermediate variable as well:
```
require(_minDisplayPrice < _maxDisplayPrice);
uint256 _completeSetCost = _maxDisplayPrice - minDisplayPrice;
require(1 <= _completeSetCost && _completeSetCost <= 2^254);
```
Also, you can drop the check for `<=2^254`.  That was just to deal with the fact that Serpent didn't have unsigned ints.  In general, anywhere we are checking to see if a number is less than 2^254 can be deleted and you should just use `uint256` instead.  Anywhere we check that a number is >= 0 can similarly just be done with types.